### ISSUE TITLE
Support for PNPM Catalogs (`catalog:` workspace protocol)

### ISSUE THREAD

**Issue by JulianCataldo** [lerna-lite/lerna-lite#902]
### Describe the bug

https://github.com/lerna/lerna/issues/4055

### Expectation

When publishing, should handle `catalog:` fields like the `workspace` ones.

### Reproduction

https://pnpm.io/catalogs

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

### Environment Info

```
  System:
    OS: macOS 14.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
    Memory: 121.36 MB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.2.0 - ~/Library/pnpm/node
    Yarn: 1.22.19 - ~/Library/pnpm/yarn
    npm: 10.7.0 - ~/Library/pnpm/npm
    pnpm: 9.5.0 - ~/Library/pnpm/pnpm
    bun: 1.0.25 - ~/.bun/bin/bun
  Browsers:
    Brave Browser: 126.1.67.134
    Safari: 17.3
  npmPackages:
    @lerna-lite/cli: ^3.7.1 => 3.7.1 
    @lerna-lite/list: ^3.7.1 => 3.7.1 
    @lerna-lite/publish: ^3.7.1 => 3.7.1 

```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/lerna-lite/lerna-lite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/lerna-lite/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/lerna-lite/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/lerna-lite/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
This was already asked in Discussion #901 and my answer is the same

> No I don't have any plans to support this and I don't think the goal would be to support every protocols that comes out. It makes sense to support the `workspace:` protocol but I really don't see why this new catalog should be supported which is a pnpm specific. On the other hand, `workspace:` was at least something that Yarn also supports and it is much more used and has been available for years (as opposed to this super new pnpm only catalog).
> 
> This new protocol also doesn't seem to be a 1-to-1 relationship like `workspace:` does, it's much more complex to support and I don't intend to support it at all. Like I said in previous paragraph, supporting every package managers little nuances is not the goal of this project

**Comment by JulianCataldo** [comment]
> It doesn't mean that I won't accept Pull Request for this feature, but I personally won't spend time on implementing it.

Hello, thanks for you prompt response, and for keeping this project maintained!
I've overlooked the discussions section, I just did a research in the Issues, sorry for that.

I'm getting familiar with the Lerna-lite repo., I'll see if I'm able to open a PR for the `catalog:` protocol.

Bye

**Comment by JulianCataldo** [comment]
For anyone in the same situation, I think using an agnostic solution that play nice with Lerna could be something like this: https://www.npmjs.com/package/syncpack

I think `catalog:` is still very new and need better awareness amongst developers, PNPM is increasingly used by a fair amount of them. Maybe if NPM or Yarn, Bun, etc. starts to support their own `catalog:` protocol, it will make this even more appealing for a wider audience to add this feature to Lerna/Lerna-lite.

**Comment by ghiscoding** [comment]
The `catalog:` protocol is now available in Lerna-Lite [v3.11](https://github.com/lerna-lite/lerna-lite/releases/tag/v3.11.0) thanks to @wassim-ben-amor for providing the code and another thanks to @Mister-Hope for opening a PR for the feature 🚀. Also note that it comes with certain limitation as it was mentioned in the PR

> The `catalog:` protocol ([pnpm catalog](https://pnpm.io/catalogs)) can be recognized by Lerna-Lite. When publishing, they will be kept as is. If you need to bump the version of a package in a catalog, you will need to edit `pnpm-workspace.yaml` manually. So we suggest using workspace: protocol instead for workspace dependencies.

**Comment by ghiscoding** [comment]
I just published [v3.12.0](https://github.com/lerna-lite/lerna-lite/releases/tag/v3.12.0) which brings full [pnpm catalogs](https://pnpm.io/catalogs) support to both `lerna version` **and** `lerna publish` commands.
